src/main/java/com/thoughtworks/gauge/datastore/DataStoreFactory.java

```diff
@@ -30,21 +41,47 @@ protected DataStore initialValue() {
      * @return The current instance of the SuiteDataStore
      */
     public static DataStore getSuiteDataStore() {
+        if (isMultithreadingExecution()) {
+            throw new RuntimeException("DataStoreFactory cannot be used for multithreaded execution. Use SuiteDataStore.");
```

Comment:
We're prepared for this but I hope it's not annoy a lot of other users. Could this just be a log line to start with? the release after can then force it? @sriv what do you think?